stdlib/public/core/Tuple.swift.gyb

```diff
@@ -0,0 +1,45 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2015 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// Generate comparison functions for tuples up to some reasonable arity.
+
+% for arity in range(2,7):
+%   typeParams = [chr(ord("A")+i) for i in range(arity)]
+%   tupleT = "({})".format(",".join(typeParams))
+
+%   equatableTypeParams = ", ".join(["{}: Equatable".format(c) for c in typeParams])
+
+@warn_unused_result
+public func == <${equatableTypeParams}>(lhs: ${tupleT}, rhs: ${tupleT}) -> Bool {
+%   ops = ["lhs.{} == rhs.{}".format(i,i) for i in range(arity)]
+  return ${" && ".join(ops)}
+}
+
+@warn_unused_result
+public func != <${equatableTypeParams}>(lhs: ${tupleT}, rhs: ${tupleT}) -> Bool {
+%   ops = ["lhs.{} != rhs.{}".format(i,i) for i in range(arity)]
+  return ${" || ".join(ops)}
+}
+
+%   comparableTypeParams = ", ".join(["{}: Comparable".format(c) for c in typeParams])
+%   for op in ["<", ">"]:
+%     for opeq in ["", "="]:
+@warn_unused_result
+public func ${op}${opeq} <${comparableTypeParams}>(lhs: ${tupleT}, rhs: ${tupleT}) -> Bool {
+%   for i in range(arity-1):
+  if lhs.${i} != rhs.${i} { return lhs.${i} ${op} rhs.${i} }
+%   end
+  return lhs.${arity-1} ${op}${opeq} rhs.${arity-1}
```

Comment:
As a code size optimization, you could try defining one operators in terms of others.